core/src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -173,40 +174,49 @@ trait PostgresPartitions extends BaseByteArrayJournalDao {
 
   override protected def writeJournalRows(xs: Seq[JournalRow]): Future[Unit] = {
     // Write atomically without auto-commit
-    super.writeJournalRows(xs).recoverWith {
-      case ex: SQLException if ex.getSQLState == PostgresErrorCodes.PgCheckViolation && isPostgresDriver =>
-        logger.info(s"Adding missing journal partition...")
-        attachJournalPartition(xs).flatMap(_ => super.writeJournalRows(xs))
-    }
+    attachJournalPartition(xs).flatMap(_ => super.writeJournalRows(xs))
```

Comment:
What's the impact on performance with checking partitions before each write? Did you run perf tests?